### PR TITLE
BUG: hidden ticklabels with sharex and secondary

### LINKED ISSUE #9158: x tick labels not shown when using secondary_y=True on figure with multiple subplots
This worked on 0.15.1, but not anymore on 0.15.2:
```python
import pandas as pd
d = {'A' : [1., 2., 3., 4.], 'B' : [4., 3., 2., 1.], 'C': [5, 1, 3, 4]}
df = pd.DataFrame(d, index=pd.date_range('2014 10 11', '2014 10 14'))

axes = df[['A','B']].plot(subplots=True)
df['C'].plot(ax=axes[0], secondary_y=True)   # x tick labels dissappear when this line is executed
```
No x tick labels are shown when using secondary_y on a subplots within a figure containing multiple subplots.
secondary_y works well when only using one subplot (subplots=False) or when layouting the subplots in columns instead of rows.

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.9.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 69 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.15.2
nose: 1.3.4
Cython: 0.21.1
numpy: 1.9.1
scipy: 0.14.0
statsmodels: 0.6.1
IPython: 2.3.1
sphinx: 1.2.3
patsy: 0.3.0
dateutil: 2.3
pytz: 2014.10
bottleneck: None
tables: 3.1.1
numexpr: 2.4
matplotlib: 1.4.2
openpyxl: None
xlrd: 0.9.3
xlwt: None
xlsxwriter: 0.6.4
lxml: 3.4.1
bs4: 4.3.2
html5lib: None
httplib2: None
apiclient: None
rpy2: 2.5.2
sqlalchemy: 0.9.8
pymysql: None
psycopg2: None
```

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pydata/pandas/issues/9158

``` python
import pandas as pd
d = {'A' : [1., 2., 3., 4.], 'B' : [4., 3., 2., 1.], 'C': [5, 1, 3, 4]}
df = pd.DataFrame(d, index=pd.date_range('2014 10 11', '2014 10 14'))

axes = df[['A','B']].plot(subplots=True)
df['C'].plot(ax=axes[0], secondary_y=True)   # x tick labels dissappear when this line is executed
```

gh

Introduced in the subplot refactoring in https://github.com/pydata/pandas/pull/7457

cc @sinhrks

The fix is pretty simple, but I'm worried I may have broken other things. No tests failed though, so hopefully we're good.

### CODE REVIEW

File: doc/source/whatsnew/v0.16.0.txt [hunk 1]
```diff
@@ -138,3 +138,8 @@ Bug Fixes
 - DataFrame now properly supports simultaneous ``copy`` and ``dtype`` arguments in constructor (:issue:`9099`)
 - Bug in read_csv when using skiprows on a file with CR line endings with the c engine. (:issue:`9079`)
 - isnull now detects NaT in PeriodIndex (:issue:`9129`)
+
+
+
+- Bug in hiding ticklabels with subplots and shared axes when adding a
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.0.txt]: no spaces needed on multiple lines (it will auto-wrap)

**Comment by jreback** [comment]
looks good. minor doc issue.

**Comment by jreback** [comment]
@TomAugspurger looks ok, pls rebase.

**Comment by jreback** [comment]
needs a rebase

**Comment by TomAugspurger** [comment]
Rebased... will merged later today.

**Comment by TomAugspurger** [comment]
The failure [here](https://travis-ci.org/pydata/pandas/jobs/59048983) was a build failure on a job that doesn't run the plotting code. Going to merge

**Comment by TomAugspurger** [comment]
Well, I messed that up. Pushed before `--continuing` my rebase, and it looks like github has it's done with this PR

**Comment by shadowleaves** [comment]
this problem seems to be still existing in pandas 0.17.0, on subplots with at least two rows:

```python
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt

mat = np.random.rand(10,2)
df = pd.DataFrame(mat, columns=['A','B'])

fig, axes = plt.subplots(nrows=2, ncols=2)                                 
df.plot(secondary_y='B', ax=axes[0, 1], legend=False)
```

image

**Pull request closed without merging**